pkg/database/database.go

```diff
@@ -132,7 +147,21 @@ func (db *Database) OpenWithCacher(ctx context.Context, cacher cache.Cacher) err
 	var rawDB *gorm.DB
 	if err := withRetries(ctx, func(ctx context.Context) error {
 		var err error
-		rawDB, err = gorm.Open("postgres", c.ConnectionString())
+		driver := ocsql.Wrap(&postgres.Driver{})
+		if !stringInSlice(driverName, sql.Drivers()) {
+			ocsql.RegisterAllViews()
+			sql.Register(driverName, driver)
+		}
+		dbSQL, err := sql.Open(driverName, c.ConnectionString())
```

Comment:
Ah yes, this bug 😄 

This is happening in a retry loop (see L148). It's possible that the sql connection successfully opens, but then gorm fails. This would trigger the retry loop to run again, creating a new sql connection. If gorm continues to fail, we'd have ~14 open sql connections multiplied by N services and could exhaust connections.

We should move the `sql.Open` call into it's own `withRetries` block and wrap it as retryable:

```go
var rawSQL *sql.DB
if err := withRetries(ctx, func(ctx context.Context) error {
  var err error
  rawSQL, err = sql.Open(driverName, c.ConnectionString())
  if err != nil {
    return retry.Retryable(err)
  }
}); err != nil {
  return fmt.Errorf("failed to create sql connection: %w", err)
}
```

Reply:
Done